pkg/scheduler/queue/tenant_queues.go

```diff
@@ -154,31 +134,27 @@ func (q *tenantQueues) getNextQueueForQuerier(lastUserIndex QueueIndex, querierI
 		return nil, "", uid
 	}
 
-	for iters := 0; iters < len(q.tenants); iters++ {
+	for iters := 0; iters < q.mapping.Len(); iters++ {
 		uid = uid + 1
 
-		// Don't use "mod len(q.users)", as that could skip users at the beginning of the list
-		// for example when q.users has shrunk since last call.
-		if int(uid) >= len(q.tenants) {
+		if int(uid) >= q.mapping.Len() {
 			uid = 0
 		}
 
-		u := q.tenants[uid]
-		if u == "" {
-			continue
+		tq := q.mapping.Get(uid)
```

Comment:
Just so I understand. The iteration order needs to be kept so that the scheduling is deterministic, ie the next queue is is fixed by the order, right?

Reply:
Right